src/internal/pfsdb/commits.go

```diff
@@ -521,10 +521,21 @@ func UpdateCommit(ctx context.Context, tx *pachsql.Tx, id CommitID, commitInfo *
 		Error:          commitInfo.Error,
 	}
 	query := updateCommit
-	_, err := tx.NamedExecContext(ctx, query, update)
+	res, err := tx.NamedExecContext(ctx, query, update)
 	if err != nil {
 		return errors.Wrap(err, "exec update commitInfo")
 	}
+	rowsAffected, err := res.RowsAffected()
+	if err != nil {
+		return errors.Wrap(err, "rows affected")
+	}
+	if rowsAffected == 0 {
+		_, err := GetRepoByName(ctx, tx, commitInfo.Commit.Repo.Project.Name, commitInfo.Commit.Repo.Name, commitInfo.Commit.Repo.Type)
+		if err != nil {
+			return err
```

Comment:
I think in these cases where we check the existence of an upper level resource, it's still good to include a commit not found error. So maybe `errors.Join(&CommitNotFoundError{RowID: id}, err)`? Fundamentally, the truth is that the commit isn't found, we are just adding more information to the error.

Reply:
that's a good idea